modules/layers/src/text-layer/text-layer.js

```diff
@@ -74,9 +86,20 @@ export default class TextLayer extends CompositeLayer {
     }
   }
 
-  updateFontAtlas(fontFamily, characterSet) {
+  updateFontAtlas() {
+    const startTime = Date.now();
+
     const {gl} = this.context;
-    const {scale, mapping, texture} = makeFontAtlas(gl, {fontFamily, characterSet});
+    const {sdf, fontFamily, characterSet} = this.props;
+
+    const {scale, mapping, texture} = makeFontAtlas(gl, {
+      sdf: sdf ? SDFPreset : null,
```

Comment:
- User settings should be merged with the preset
- Override `fontFamily` with the value from props